ldms/src/ldmsd/ldms_ls.c

```diff
@@ -728,7 +728,14 @@ void fprint_decomp(FILE *fp, ldms_set_t s)
 			enum ldms_value_type rtype;
 			ldms_mval_t lval = ldms_metric_get(s, i);
 			ldms_mval_t rval = ldms_list_first(s, lval, &rtype, NULL);
-			fprint_record(fp, 14, ldms_metric_name_get(s, i), rval);
+			if (rval) {
+				fprint_record(fp, 14, ldms_metric_name_get(s, i), rval);
+			} else {
+				/* Can't do decomp for an empty list */
+				fprintf(fp, "              { \"src\" : \"%s\", "
+					"\"dst\" : \"ERROR: Cannot generate decomposition for an empty list\" }",
```

Comment:
@nichamon, in my opinion "no" because the decomposition file generation is advisory. There is nothing _wrong_ with an empty list, it's just that the decomposition logic can't advise you how to decompose a list record it doesn't have.